### ISSUE TITLE
Calendar Widget

### ISSUE THREAD

**Issue by mpfau** [tutao/tutanota#1756]
As a user i want to be able to install a tuta calendar widget on the homescreen so that i can see upcomming events without opening the app.

### AC
* A widget for each calendar view is availble for installation.
* The widget contains a button to create a new event.
* There is an option to select the calendars that are available.

### Notes
We should introduce this after we have released the calendar app.

---
**Updated Discussion**
As a user of a mobile app i want to be able to add a calendar widget on the home-screen of my device so i can see the next upcoming events.

**AC v1**
**Widget Configuration**
- User can chose a account to fetch data from
- User can chose which calendars to display the events

**Layout**
We support different sizes:
- Android, re-sizable with minimum 3x2 (width/height)
- iOS, 4x4 (width/height)
- The vertical layout is applied for
  - iOS 4x4 and 2x2
  - All layouts that are not Android 4x2 
- The widget should display the day number, month
- We show all the upcoming events of the current day including all day events.
  - All day events are summarized at the top to avoid using too much space.
    - When there is only one day event the event title is shown
    - When there are multiple all day events only the amount of all day events are shown.
  - On Android,: The list of upcomming events should be scrollable.
  - On iOS: Have a text on the bottom if there's more events (iOS doesn't support scrolling)
- There should be a button to create a new event.
  - Button should be hidden if the device is too narrow, prioritizing displaying information.

**Interactions** 
  - When user clicks on the day the app opens the agenda view
  - When user clicks an event it should open the agenda view and open the event preview.
  - If user taps add button, opens the app on the Event Editor
- The data should update automatically.

**Comment by Joeyzo** [comment]
Really in need for this feature as well!

**Comment by 2br-2b** [comment]
Yes!  I want this feature a lot!

**Comment by laius** [comment]
I agree!

**Comment by jtw023** [comment]
Definitely agree! It would be amazing to have the ability to integrate the tutanota calendar with linux, ios, and android operating systems rather than having to go into the email client first. I hope this one comes to fruition!

**Comment by akaessens** [comment]
> Definitely agree! It would be amazing to have the ability to integrate the tutanota calendar with linux, ios, and android operating systems rather than having to go into the email client first. I hope this one comes to fruition!

Hello, an integration into various systems would require something like webdav sync but that's not e2e-encrypted (see [here](https://mastodon.social/@Tutanota/102404709788706687)).
When this comes, i expect it to be implemented into the email client (or call it email+calendar client? ;) ). I think it also depends on the offline mode to be more useful.

Greetings from Hanover to the Team!

**Comment by IrakliXYZ** [comment]
![widget](https://user-images.githubusercontent.com/39024518/99558357-144c7b80-29dd-11eb-95a0-c84d928e8bbd.png)
Here is what it can look like. It can be bit tricky to implement, if you are gonna add the fingerprint/pin security feature too. Maybe the widget won't show the event details when you click on it, but it will open the app and require you to authenticate.

**Comment by yann-soubeyrand** [comment]
If a contact and calendar providers (as discussed in https://github.com/tutao/tutanota/issues/1066) are implemented, one could use any calendar widget (like the one provided out of the box with most smartphones). It would also unlock other use cases like dialer integration, the possibility to add or join a Jitsi Meet link associated to a meeting, the ability to see one's contacts in other messaging apps like Signal, etc. Maybe these contact and calendar providers should be optionally activated so that the most security concerned persons could leave it deactivated though.

**Comment by freedriven** [comment]
YES!  As mentioned in #1066 , the option to integrate contacts and calendars is critical.  Encryption and privacy should always be an option, even an easy option, but not a mandate. Of course, a long term goal of e2ee for all information is great. But there is always the trade-off of ease of use and security. And as of now, Tutanota is forcing me to choose full security at a high cost of ease of use, and I choose otherwise.

**Comment by yann-soubeyrand** [comment]
@freedriven don't get me wrong, I'm not saying that E2EE should be optional: this is the raison d'être of Tutanota and I hope it'll always be ;-) What I say is that a lot of features can be unlocked if we consider that the user end in E2EE is the phone. However, I think that the user should be able to chose if she trusts her phone to be the user end or if she considers that the user end should be the Tutanota app and nothing else.

**Comment by freedriven** [comment]
Yea, my post was written very poorly. I didn't mean to say that Tutanota _email_ E2EE should be optional. I agree, that's the service they provide and core mission.

To put it another way, as is, I won't use Tutanota.  But, if they integrated user/phone Calendar and Contact data into their app, even if that data was handled using industry standard security protocols and not encrypted at rest on Tutanota servers, I'd switch to Tutanato.  That's an ease of use vs security trade-off I'd personally make.

**Comment by adguardgirl** [comment]
God I'm missing the calendar widget .(

**Comment by rafaeldrs** [comment]
Can't agree more with others. This is a must have feature. Anyway, thanks for everything tutanota team. Love the service 👏

**Comment by elairat** [comment]
Seriously, if this gadget doesn't show up asap, I will move to protonmail. I've been waiting for this now more than a year because I so believe in Tutanota and love all the rest they do. But I can't run my business without this one. This gadget is a must! And I don't want to be connected to my mails all the time, it's crazy on my brain overload.

**Comment by antoniotari** [comment]
any news about this? This is pretty essential for Android users and I can see this thread has been open since jan 2020. In my opinion calendar should be a high priority feature. Please let us know any update about the expected delivery date for it. Thanks.

**Comment by Gnorv** [comment]
Here to give a +1 for this issue. Since getting Tutanota I am waiting to ditch other calendar apps on my phone but without fast calendar access Tutanotas calendar is just unusable for me. I prefer a clean widget with upcoming events to clicking 5 buttons and waiting for an app to load until I can see my entries.
Even better if the calendar would integrate into Linux/Ubuntu so I can use its native calendar.

**Comment by valtoree** [comment]
looking forward to having this implemented, I know it's on the road map. hopefully sooner rather than later. And monthly view pls 🙏

**Comment by dermitdemdino** [comment]
A lot of time has passed, unfortunately a widget is not yet in sight.

**Comment by iim711** [comment]
I have just signed up to Tuta. This is the top feature I'm missing.

**Comment by valtoree** [comment]
Anytime now.... They've made some calendar improvements so maybe soon 🤷🤞

**Comment by IPv777** [comment]
And now ? Any update please....

**Comment by alexander-n8hgeg5e** [comment]
I really need this feature. I need to see upcoming calendar entries on the homescreen somehow.

**Comment by antebudimir** [comment]
So, we recently got a calendar app (that provides everything the Tuta mail app provides 🙈), but we still don't have a widget. Is there any progress on the widget, please?

**Comment by jacrook** [comment]
Second @antoniotari  comment

Is there a beta version that has a widget to help test?

**Comment by eclecticpassions** [comment]
@antoniotari +1. Any updates on the Android widgets? 

Also any chance for a Schedule view like in Google calendar, showing all the future upcoming events in order of date? Not like the Agenda view currently offered on Tuta that only shows the list of events limited for one day.

**Comment by eclecticpassions** [comment]
@antoniotari +1. Any updates on the Android widgets? 

Also any chance for a Schedule view like in Google calendar, showing all the future upcoming events in order of date? Not like the Agenda view currently offered on Tuta that only shows the list of events limited for one day.

Thanks!

**Comment by RDavis3000** [comment]
If i had known this didn't exist I would not have subscribed to the product. How about just allowing regular mail/calendar clients to pull it? TBH i dont really care about encryption

**Comment by alexander-n8hgeg5e** [comment]
@RDavis3000 It doesn't work like that, you need to subscribe in advance and then
money -> pizza, candy -> coding

**Comment by dodo-the-dev** [comment]
I need this. Calendar widget is one of the most used widgets on my phone (Android). Please make it a month view, to make it better than Proton (who I just moved from and only had an agenda view widget).

**Comment by Nun-Of-Ur** [comment]
Tuta has been saying a widget is coming for years. They are very quiet about it now. Apparently it is not a priority for them, or there is some technical issue that makes it to difficult to implement. The silence from them is the bad part, just say what the issue is.

**Comment by eclecticpassions** [comment]
Yeah, I agree it's been a long time... Hate to say it but it is a big deciding factor when it comes to renewal time.

**Comment by charlag** [comment]
Hi, it is planned and is a priority for Calendar team! Sorry for the long wait, we understand that it is frustrating!

**Comment by dustingrady** [comment]
Very much looking forward to this!

**Comment by luc-spec** [comment]
+1 for this ask. It would be a huge quality of life improvement! Glad to hear the team is working on it!

**Comment by charlag** [comment]
There was an Android app version released yesterday that includes calendar widget!

**Comment by alexander-n8hgeg5e** [comment]
> There was an Android app version released yesterday that includes calendar widget!

Just installed it. It's working and it looks good !  :-)

**Comment by mjshonty** [comment]
> > There was an Android app version released yesterday that includes calendar widget!
> 
> Hi! I can't see it 🤔

It doesn't seem like F-Droid or the Play Store have updated with the new build just yet. I went to the releases on GitHub and installed the APK directly. It was at the bottom of the page here: https://github.com/tutao/tutanota/releases/tag/tuta-calendar-android-release-277.250409.0

**Comment by dustingrady** [comment]
Great job on this, Tuta Team!

**Comment by 5onderling** [comment]
I also have some problems with the widget not showing any events when there are some. I have to click on the day a few times to open in the calendar app and then at some point the widget shows a loading spinner and finally displays my events. So I sadly can't really trust it as it is.

**Comment by jm355** [comment]
I'm also having this issue. At first it didn't show any events, except newly added ones. I also tried editing an event and then saving it and that fixed it for about half of the events, but others are still missing

**Comment by jm355** [comment]
Aso had an event disappear from the widget before the event was over. Opening and closing tuta calendar until the widget refreshed caused it to reappear

**Comment by valtoree** [comment]
Ty so much devs for widget ❤️
I still can't use it, NEED month view 😪
Oh well.... Maybe one day...

**Comment by alexander-n8hgeg5e** [comment]
Is the (agenda) view of the widget meant to show only the events from the current day ? 
For me it needs to show all the upcoming events that fit into the widget.
So I always see what's upcoming and if the widget is full I know I have to open the app to see the full extent.

**Comment by luc-spec** [comment]
For the team's awareness, I do not have a widget option on GrapheneOS. I understand that only mainline Android is supported. Main goal here is to add the data point and potentially inform other GOS users.

Tuta version: 277.250414.1 (source GitHub / Obtainum)
GOS build: 2025042500

**Comment by ymstnt** [comment]
> For the team's awareness, I do not have a widget option on GrapheneOS. I understand that only mainline Android is supported. Main goal here is to add the data point and potentially inform other GOS users.
> 
> Tuta version: 277.250414.1 (source GitHub / Obtainum) GOS build: 2025042500

Which app are you using? The widget is only available in Tuta Calendar, not the normal Tuta (mail) app.

**Comment by GoCho85** [comment]
> Is the (agenda) view of the widget meant to show only the events from the current day ? For me it needs to show all the upcoming events that fit into the widget. So I always see what's upcoming and if the widget is full I know I have to open the app to see the full extent.

AFAIK agenda means only events for today.
But I'd like to see all upcoming events either, so maybe the widget can get this function as well to choose from. And even shows today pasts events.
just a humble suggestion from me

**Comment by GoCho85** [comment]
> For the team's awareness, I do not have a widget option on GrapheneOS. I understand that only mainline Android is supported. Main goal here is to add the data point and potentially inform other GOS users.
> 
> Tuta version: 277.250414.1 (source GitHub / Obtainum) GOS build: 2025042500

The calendar app incl the updated widget works for me on GrapheneOS.

**Comment by santiagodmnch** [comment]
This issue has been open for 5 years. How isn’t there a widget for iOS yet?

**Comment by charlag** [comment]
It's about to be released :)

**Comment by Backspaze** [comment]
A while back, I raised a question on Reddit regarding support for a [iOS lock screen widget](https://support.apple.com/en-us/118610#lock-screen), and got the [following response](https://www.reddit.com/r/tutanota/comments/d0at9u/comment/lq9ivwk):

> Thanks for clarifying, I will raise this question with our development team and let you know when I have more info.
> 
> Update: Our team has included your lock screen request in their planning for the widget functionality.

I see no mention of a lock screen widget in this issue, and I can't find any other GitHub issue regarding this. Are there any plans to add support for a lock screen widget in this issue, or should I post another feature request on Reddit?

**Comment by sf985** [comment]
Hello, first I'd like to say that I appreciate your apps and services and I'd just like to add my vote to the possibility of setting the widget (or a choice of widget type) to show upcoming week, month, all events, etc. -perhaps in line with the options inside the calendar app. Me and many others don't work (or don't work only) with intra-day tasks, but inter-day and over different time periods like a week, weeks, months, and some that simply work by whatever time period is included within an event end date (whether it be e.g. a vacation or a work project). Thank you for the consideration!

**Comment by hex0lon** [comment]
Hello, a MacOS Widget would be great!
